cpp/src/distance/distance-ext.cuh

```diff
@@ -133,13 +133,16 @@ void distance(raft::resources const& handle,
               raft::device_matrix_view<OutT, IdxT, layout> dist,
               DataT metric_arg = 2.0f) RAFT_EXPLICIT;
 
-template <typename Type, typename layout = layout_c_contiguous, typename IdxT = int>
+template <typename Type,
+          typename layout = layout_c_contiguous,
+          typename IdxT   = int,
+          typename DistT  = Type>
```

Comment:
I'm really really not in love at all with introducing new template types. We should be going the opposite direction.